internal/authentication/azure/metadata-properties.go

```diff
@@ -35,6 +35,9 @@ var MetadataKeys = map[string][]string{ //nolint:gochecknoglobals
 	// Identifier for the Azure environment
 	// Allowed values (case-insensitive): AzurePublicCloud/AzurePublic, AzureChinaCloud/AzureChina, AzureUSGovernmentCloud/AzureUSGovernment
 	"AzureEnvironment": {"azureEnvironment", "azureCloud"},
+	// Identifier for the Azure authentication methods to try (in order), comma-separated
+	// Allowed values (case-insensitive): ServicePrincipal, Certificate, WorkloadIdentity, ManagedIdentity, CLI, None
```

Comment:
I think "Service Principal" and "Certificate" are bad names. All the methods above use a service principal (also MI uses a SP).

More appropriate IMHO would be "ClientCredentials" and "ClientCertificate"